pandas-stubs/core/indexes/multi.pyi

```diff
@@ -27,31 +29,42 @@ from pandas._typing import (
 class MultiIndex(Index[Any]):
     def __new__(
         cls,
-        levels=...,
-        codes=...,
-        sortorder=...,
+        levels: SequenceNotStr[SequenceNotStr[Hashable] | AnyArrayLike] = ...,
+        codes: SequenceNotStr[SequenceNotStr[int]] = ...,
```

Comment:
```suggestion
        codes: Sequence[Sequence[int]] = ...,
```
We know it has to be a sequence of sequence of int - don't need `SequenceNotStr` here.

We only use that when we want to allow a sequence that is specifically not a specific string.